authentication/python/azure/authentication_azure/azure_identity_access_token_provider.py

```diff
@@ -0,0 +1,55 @@
+from typing import Dict, List, Optional, Set
+
+from kiota.abstractions.authentication import AccessTokenProvider, AllowedHostsValidator
+
+from azure.core.credentials_async import AsyncTokenCredential
+
+
+class AzureIdentityAccessTokenProvider(AccessTokenProvider):
+    """Access token provider that leverages the Azure Identity library to retrieve an access token.
+    """
+
+    def __init__(
+        self,
+        credentials: AsyncTokenCredential,
+        options: Optional[Dict],
+        scopes: List[str] = ['https://graph.microsoft.com/.default'],
+        allowed_hosts: Set[str] = {
+            'graph.microsoft.com', 'graph.microsoft.us', 'dod-graph.microsoft.us',
+            'graph.microsoft.de', 'microsoftgraph.chinacloudapi.cn', 'canary.graph.microsoft.com'
+        },
+    ) -> None:
+        if not credentials:
+            raise Exception("Parameter credentials cannot be null")
+        if not scopes:
+            raise Exception("Scopes cannot be null")
+
+        self._credentials = credentials
+        self._scopes = scopes
+        self._options = options
+        self._allowed_hosts_validator = AllowedHostsValidator(allowed_hosts)
+
+    async def get_authorization_token(self, uri: str) -> str:
+        """This method is called by the BaseBearerTokenAuthenticationProvider class to get the
+        access token.
+        Args:
+            uri (str): The target URI to get an access token for.
+        Returns:
+            str: The access token to use for the request.
+        """
+        if not uri or not self.get_allowed_hosts_validator().is_url_host_valid(uri):
```

Comment:
missing a check for https

Reply:
Thanks. Added in https://github.com/microsoft/kiota/pull/1108/commits/0e5a3daee7e53f5ca03e17913644656d5e8e4ee7